operators/src/util/raster_stream_to_png.rs

```diff
@@ -25,8 +28,21 @@ where
 {
     let colorizer = colorizer.unwrap_or(default_colorizer_gradient::<T>()?);
 
+    let valve_trigger = query_ctx.valve_trigger();
+
     let tile_stream = processor.query(query_rect, &query_ctx).await?;
 
+    conn_close.map(|c| {
+        crate::util::spawn(async move {
+            if c.await.is_ok() {
+                if let Some(trigger) = valve_trigger {
+                    info!("shoot trigger");
+                    trigger.cancel();
+                }
+            }
+        })
+    });
```

Comment:
Can this future maybe implement Abortable and then cancel it on being aborted?

I'm unsure about the pistol and trigger terminology. Maybe we should stay with stream and cancel :wink: .

Reply:
what future do you mean? `conn_close` shouldn't be aborted, it finishes when the connection is aborted and then cancels the query processors.